<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">

    <title>What is Cytoscape?</title>

    <meta name="description" content="An Overview of Cytoscape Project.">
    <meta name="author" content="Keiichiro Ono">

    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <script src="https://code.jquery.com/jquery-2.1.4.min.js"></script>

    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css" integrity="sha512-dTfge/zgoMYpP7QbHy4gWMEGsbsdZeCXz7irItjcC3sPUFtf0kuFbDz/ixG7ArTxmDjLXDmezHubeNikyKGVyQ==" crossorigin="anonymous">

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js" integrity="sha512-K1qjQ+NcF2TYO/eI3M6v8EiNYZfA95pQumfvcVrTHtwQVDG+aHRqLi/ETn2uB+1JqwYqVG3LIvdm9lj6imS/pQ==" crossorigin="anonymous"></script>

    <link href="css/main.bootstrap.css" rel="stylesheet">

    <script src="js/cytoscape.js-2.5.0/cytoscape.min.js"></script>

    <!-- For Scroll -->
    <script src="js/smooth-scroll.min.js"></script>

</head>

<body>

    <div class="white image-wide">
        <img src="images/3_1_title3.png" />
    </div>

    <div class="belt light-gray">
        <div class="container">
            <div class="row">
                <div class="col-sm-12 col-md-4">
                    <img src="images/logo/cy3logoOrange.svg" width="100%" style="margin-top: 2em;" />
                </div>
                <div class="col-sm-12 col-md-8">
                    <h1>What is Cytoscape?</h1>

                    <p class="lead">
                        <strong><i>Cytoscape</i></strong> is an open source software platform for <strong><i>
                    visualizing</i></strong> molecular interaction networks and biological pathways and
                        <strong><i>integrating</i></strong> these networks with annotations, gene expression profiles and other state data. Although Cytoscape was originally designed for biological research, now it is a general platform for complex network
                        analysis and visualization. &nbsp; Cytoscape <i>core</i> distribution provides a basic set of features for data integration, analysis, and visualization. &nbsp; Additional features are available as
                        <strong><i>Apps</i></strong> (formerly called <i>Plugins</i>). &nbsp; Apps are available for network and molecular profiling analyses, new layouts, additional file format support, scripting, and connection with databases. &nbsp;
                        They may be developed by anyone using the Cytoscape open API based on <a href="http://www.java.com/en/" target="_blank">Java&#8482;</a> technology and App community development is encouraged. Most of the Apps are freely available
                        from <a href="http://apps.cytoscape.org/">Cytoscape App Store</a>.
                    </p>
                </div>
            </div>
            <div class="centering">
                <a href="#section-image1" data-scroll>
                    <h2><span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span></h2>
                </a>
            </div>
        </div>
    </div>

    <div id="section-image1" class="belt black image-wide">
        <div class="container">
            <img src="images/screenshots/structure-aware-layout.jpg" style="width: 100%" />
            <div class="centering">
                <a href="#section-family" data-scroll>
                    <h2><span class="glyphicon glyphicon-menu-down" aria-hidden="true" style="color: white;"></span></h2>
                </a>
            </div>
        </div>
    </div>


    <div id="section-family" class="belt white cy3-intro">
        <div class="container">
            <div class="centering">
                <h1>Cytoscape Family</h1>
            </div>
            <div class="row">
                <div class="col-xs-12 col-md-6">
                    <h2>Cytoscape 3</h2>

                    <p class="lead">
                        <strong><i>Cytoscape 3</i> is the mainstream version of Cytoscape with modular architecture</strong>. It is designed for long-term maintainability and it replaced 2.x series. It is a Java desktop application designed for large-scale
                        network analysis and visualization. New features, including new user interfaces, advanced visualization functions, headless (command-line) distribution, RESTful API, and multiple rendering engine support, will be released for this
                        version. <strong>As of May 2016, transition from Cytoscape 2.x to 3.x is complete, and 2.x is no longer supported.</strong> If you really need to use Plugins only available for 2.x, you may use Cytoscape 2.8.3, but otherwise, please
                        use the latest version of Cytoscape 3.x series.

                    </p>
                    <p class="lead">
                        Please read <a href="roadmap.html">roadmap page</a> for further information.
                    </p>
                </div>
            </div>
            <div class="centering">
                <a href="#section-cy3-desktop" data-scroll>
                    <h2><span class="glyphicon glyphicon-menu-down" aria-hidden="true" style="color: black;"></span></h2>
                </a>
            </div>
        </div>
    </div>

    <div id="section-cy3-desktop" class="belt white image-wide">
        <div class="container centering">
            <h2>Cytoscape 3.5 Desktop</h2>
            <img src="images/screenshots/cy3_5_0_desktop.png" style="width: 100%" />
            <div class="row narrow">
                <a href="#section-cyjs" data-scroll>
                    <h2><span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span></h2>
                </a>
            </div>
        </div>
    </div>


    <div id="section-cyjs" class="narrow cyjs-back" id="js-section">
        <div class="container">
            <div class="row">
                <div class="col-xs-12 col-md-offset-6 col-md-6">
                    <h2>Cytoscape.js</h2>

                    <p class="lead">
                        <a href="https://cytoscape.github.com/cytoscape.js/index.html">Cytoscape.js</a> is a successor of
                        <a href="http://cytoscapeweb.cytoscape.org">Cytoscape Web</a>. Cytoscape.js is a <strong>JavaScript
                    library</strong> for network visualization and analysis, and is <strong>NOT a complete web
                    application</strong>. Cytoscape 3 and cytoscape.js share design-level concepts, such as Visual Styles, but their code bases are completely independent to each other. It is designed to be a building block for complex data visualization
                        web applications for the HTML5 era. The Cytoscape core developer team is trying to improve interoperability of Cytoscape 3 and Cytoscape.js. From Cytoscape 3.2, users can export network visualizations as interactive web application
                        using Cytoscape.js.
                    </p>

                    <h3 class="warn">This background is rendered by Cytoscape.js</h3>
                </div>
            </div>
        </div>
    </div>

    <div class="cyjs-back">
        <div class="cyjs">

        </div>

        <div class="centering row narrow">
            <a href="#section-cy2" data-scroll>
                <h2><span class="glyphicon glyphicon-menu-down" aria-hidden="true" style="color: white;"></span></h2>
            </a>
        </div>
    </div>

    <div id="section-cy2" class="belt light-gray" id="js-section">
        <div class="container">
            <div class="row">
                <a name="which_cytoscape"></a>

                <div class="col-xs-12 col-md-5">
                    <h3>Cytoscape 2</h3>

                    <p>
                        <strong>Cytoscape 2.x</strong> is a legacy version of Cytoscape. This version is now desupported. New features will not be added by the core developers, and it may be hard to get answers to 2.x specific questions. Please do NOT
                        use this version unless you really need to use plugins only for 2.x series.
                    </p>
                </div>
                <div class="col-xs-12 col-md-7">

                    <h3>Which one is the right one for me?</h3>

                    <ul class="list-unstyled">
                        <li>
                            <strong>Users, Casual Users, and Power Users:</strong> Use Cytoscape v3.x -- v2.x has been desupported. Use v2.x if you have used v2.x in the past and rely on a plugin not supported as a v3.x app in the App Store â€“ keep checking
                            the App Store for new arrivals.
                        </li>
                        <li>
                            <strong>App Developers:</strong> Please port your 2.x Apps to 3.x series. We recommend to use 3.x for new plugins development, as 2.x is desupported. <a href="http://chianti.ucsd.edu/cytoscape-3.4.0/API/">3.x
                        API</a> is stable enough to develop new Apps, but if you notice any design flaws or bugs,
                            <a href="https://groups.google.com/forum/?fromgroups#!forum/cytoscape-discuss">
                            please let us know</a>.
                        </li>
                        <li>
                            <strong>Web Application Developers:</strong> You can use cytoscape.js as network data visualization engine for your web application. It is a pure JavaScript library and no plugin is required for the web browsers. In the long
                            term, this library and other server-side applications replaces Java Web Start version of Cytoscape.
                        </li>
                    </ul>
                </div>
            </div>
            <div class="row centering">
                <a href="#section-features" data-scroll>
                    <h2><span class="glyphicon glyphicon-menu-down" aria-hidden="true" style="color: darkgray;"></span></h2>
                </a>
            </div>
        </div>
    </div>


    <div id="section-features" class="narrow white">
        <div class="container centering">
            <h1>Cytoscape 3 Core Features</h1>
        </div>
    </div>

    <div class="belt white">
        <div class="container">
            <div class="row">
                <div class="col-md-4 panel">
                    <h4>Supports Many Standards</h4>
                    <img src="images/intro_3.png" />

                    <p>
                        Cytoscape supports a lot of standard network and annotation file formats including: SIF (Simple Interaction Format), <a href="http://www.infosun.fim.uni-passau.de/Graphlet/GML/"> GML</a>,
                        <a href="http://cgi5.cs.rpi.edu/research/groups/pb/punin/public_html/XGMML/">
                        XGMML</a>, <a href="http://www.biopax.org/">BioPAX</a>,
                        <a href="http://www.psidev.info/index.php?q=node/60">PSI-MI</a>,
                        <a href="http://graphml.graphdrawing.org/">GraphML</a>,
                        <a href="http://www.kegg.jp/kegg/xml/">KGML (KEGG XML)</a>,
                        <a href="http://sbml.org/Main_Page">SBML</a>, <a href="http://www.obofoundry.org/">OBO</a>, and <a href="http://www.geneontology.org/GO.format.annotation.shtml">Gene Association</a>. Delimited text files and MS Excel&#8482; Workbook
                        are also supported and you can import data files, such as expression profiles or <a href="http://www.geneontology.org">GO</a> annotations, generated by other applications or spreadsheet programs. Using this feature, you can load
                        and save arbitrary attributes on nodes, edges, and networks. For example, input a set of custom annotation terms for your proteins, create a set of confidence values for your protein-protein interactions.
                    </p>
                </div>
                <div class="col-md-4 panel">
                    <h4>Public Database Clients</h4>

                    <p>
                        <img src="images/intro_4.png" />
                    </p>

                    <p>
                        Cytoscape works as a web service client. This means Cytoscape can directly connect to the external public databases and imports network and annotation data. Currently, <a href="http://www.pathwaycommons.org/pc/">Pathway Commons</a>,
                        <a href="http://www.ebi.ac.uk/intact/site/index.jsf">IntAct</a>, <a href="http://www.biomart.org/">BioMart</a>, and <a href="http://www.ncbi.nlm.nih.gov/sites/entrez?db=gene">NCBI Entrez Gene</a> are supported. And we continue
                        to develop new service clients for popular databases.
                    </p>
                </div>
                <div class="col-md-4 panel">
                    <h4>Interoperability</h4>
                    <img src="images/logo/jupyter-logo.svg" width="250px" />
                    <img src="https://www.python.org/static/community_logos/python-logo-generic.svg" width="250px" />
                    <p>
                        Since Cytoscape supports import/export standard file formats, you can easily put Cytoscape into your workflow. For example, if you have a network data generated by
                        <a href="http://cneurocvs.rmki.kfki.hu/igraph/">igraph</a> or
                        <a href="http://www.bioconductor.org/packages/release/GraphsAndNetworks.html">Bioconductor</a>, Cytoscape can load the file as a text table and you can export it in PSI-MI format for other bioinformatics tools or your own applications/scripts.
                    </p>
                    <p>
                        From version 3.3, Cytoscape supports <a href="http://apps.cytoscape.org/apps/cyrest">RESTful API</a> for programmatic access. You can use your choice of programming languages to access Cytoscape core features, such as creating
                        networks, applying layouts, or image export.
                    </p>
                </div>
            </div>
            <div class="row">
                <div class="col-md-4 panel">
                    <h4>Session File</h4>

                    <p>
                        <img src="images/cy3session.png" />
                    </p>

                    <p>
                        You can save your work by a single click. All of settings, data files, and visualizations are packed in a session file. It is called <i>Cytoscape Session (.cys) file</i>. Cytoscape Session file includes networks, attributes (for
                        node/edge/network), Desktop states (selected/hidden nodes and edges, window sizes), Properties, some plugin states, and Visual Styles.
                    </p>
                </div>
                <div class="col-md-4 panel">
                    <h4>Layout</h4>

                    <p>
                        <img src="images/intro_9.png" />
                    </p>

                    <p>
                        Layout networks in two dimensions. &nbsp;A variety of layout algorithms are available, including cyclic, tree, force-directed, edge-weight, and yFiles Organic layouts. You can also use Manual Layout tools similar to other graphics application user interface.
                    </p>
                </div>
                <div class="col-md-4 panel">
                    <h4>Image Export</h4>

                    <img src="images/logo/adobe21.svg" width="200" />

                    <p>
                        You can export networks as publishable-quality images. Supports PDF, PS, SVG, PNG, JPEG, and BMP files. Vector images (PDF and PS) can be modified by other application such as Adobe Illustrator for further enhancements.
                    </p>
                </div>
            </div>
            <div class="row">
                <div class="col-md-6 panel">
                    <h4>VizMapper</h4>

                    <p>
                        <img src="images/screenshots/31_vizmap.png" />
                    </p>

                    <p>
                        Customize network data display using powerful
                        <a href="http://wiki.cytoscape.org/Cytoscape_User_Manual/Visual_Styles">
                            <strong><i>VisualStyles</i></strong></a>. View a superposition of gene expression ratios and p-values on the network. &nbsp; Expression data can be mapped to node color, label, border thickness, or border color, etc. according
                        to user-configurable colors and visualization schemes.
                    </p>
                </div>
                <div class="col-md-6 panel">
                    <h4>Filter</h4>

                    <p>
                        <img src="images/screenshots/31_filter.png" width="100%" />
                    </p>

                    <p>
                        <strong>Filter the network</strong> to select subsets of nodes and/or interactions based on the current data. &nbsp;For instance, users may select nodes involved in a threshold number of interactions, nodes that share a particular
                        GO annotation, or nodes whose gene expression levels change significantly in one or more conditions according to p-values loaded with the gene expression data. You can create new networks from the filtering result. </p>
                </div>
            </div>
            <div class="row">
                <div class="col-md-6 panel">
                    <h4>Search</h4>

                    <p>
                        <img src="images/intro_17.png" />
                    </p>

                    <p>
                        Search target nodes and edges from
                        <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2427162/">Search</a> window.
                        <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2427162/table/T1/">
                        Lucene Syntax</a> is supported for arbitrary complex queries.
                    </p>
                </div>
                <div class="col-md-6 panel">
                    <h4>Browsing</h4>

                    <p>
                        <img src="images/intro_11.png" />
                    </p>

                    <p>
                        Zoom in/out and pan for browsing the network. Use the network manager to easily organize multiple networks. And this structure can be saved in a session file. Use the <i>bird's
                    eye view</i> to easily navigate large networks. Easily navigate large networks (100,000+ nodes and edges) by efficient rendering engine.
                    </p>
                </div>
            </div>
            <div class="row">
                <div class="col-md-4 panel">
                    <h4>Find Modules/ Clusters</h4>

                    <p>
                        <strong>Find active subnetworks/pathway modules</strong>. The network is screened against gene expression data to identify connected sets of interactions, i.e. <i>interaction subnetworks</i>, whose genes show particularly high
                        levels of differential expression. &nbsp;The interactions contained in each subnetwork provide hypotheses for the regulatory and signaling interactions in control of the observed expression changes. <strong>Find clusters (highly interconnected regions)</strong>                        in any network loaded into Cytoscape. Depending on the type of network, clusters may mean different things. For instance, clusters in a protein-protein interaction network have been shown to be protein complexes and parts of pathways.
                        Clusters in a protein similarity network represent protein families.
                    </p>

                    <p>
                        <img src="images/intro_14.png" width="100%" />
                    </p>

                    <p>
                        <img src="images/intro_15.png" width="100%" />
                    </p>
                </div>
                <div class="col-md-4 panel">
                    <div class="col-xs-12">
                        <h4>App Manager and <i>App Store</i></h4>
                        <img src="images/appstore.png">

                        <p>
                            <strong><a href="http://apps.cytoscape.org">Apps</a></strong> are available for network and molecular profile analysis. Cytoscape is a software written in Java and you can write your own App for data analysis, import, and visualization
                            by writing Java code. More Apps are available at the <a href="http://apps.cytoscape.org">Cytoscape App Store</a>. You can install most of the Apps by just one click from <i>App Manager</i>, or directly install from the App
                            Store (this is a Cytoscape 3 feature).
                        </p>
                    </div>
                    <div class="col-xs-12">
                        <h4>Export to Web</h4>

                        <p>
                            Cytoscape 3 can generate a web application using Cytoscape.js from your Cytoscape session. In addtion, you can easily export your networks as JSON files, and share with your collaborators using external web application such as
                            <a href="http://cynetshare.ucsd.edu/">cyNetShare</a>.
                        </p>
                    </div>
                </div>
                <div class="col-md-4 panel">
                    <h4>Multilingual</h4>
                    <img src="images/intro_13.png" width="100%" />

                    <p>
                        You can use your language for the data files. Most of the features in Cytoscape supports languages other than English, including Eastern Asian languages using two-byte characters.
                    </p>
                </div>
            </div>
        </div>

        <div class="centering">
            <a href="#" data-scroll>
                <h2><span class="glyphicon glyphicon-menu-up" aria-hidden="true" style="color: darkgray;"></span></h2>
            </a>
        </div>
    </div>



    <footer></footer>

    <script src="js/load_cyjs.js"></script>
    <script src="js/setup_page.js"></script>

    <script>
        smoothScroll.init({
            speed: 1000,
            easing: 'easeInOutCubic'
        });
    </script>

</body>

</html>
