<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">

    <title>What is Cytoscape?</title>

    <meta name="description" content="An Overview of Cytoscape Project.">
    <meta name="author" content="Keiichiro Ono">

    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <script src="http://code.jquery.com/jquery-2.1.1.min.js"></script>

    <link rel="stylesheet" href="http://netdna.bootstrapcdn.com/bootstrap/3.1.0/css/bootstrap.min.css">
    <script src="http://netdna.bootstrapcdn.com/bootstrap/3.3.0/js/bootstrap.min.js"></script>

    <link href="css/main.bootstrap.css" rel="stylesheet">
    <script src="js/cytoscape.js-2.3.4/build/cytoscape.min.js"></script>

</head>

<body>

<div class="white image-wide">
    <img src="images/3_1_title3.png"/>
</div>

<div class="belt light-gray">
    <div class="container">
        <div class="row">
            <div class="col-sm-12 col-md-4">
                <img src="images/logo/cy3logoOrange.svg" width="100%" style="margin-top: 2em;"/>
            </div>
            <div class="col-sm-12 col-md-8">
                <h2>What is Cytoscape?</h2>

                <p class="lead">
                    <strong><i>Cytoscape</i></strong> is an open source software platform for <strong><i>
                    visualizing</i></strong> molecular interaction networks and biological pathways and
                    <strong><i>integrating</i></strong> these networks with annotations, gene expression
                    profiles and other state data. Although Cytoscape was originally designed for
                    biological research, now it is a general platform for complex network analysis and
                    visualization. &nbsp; Cytoscape <i>core</i> distribution provides a basic set of features
                    for data integration, analysis, and visualization. &nbsp; Additional features are available as
                    <strong><i>Apps</i></strong> (formerly called <i>Plugins</i>). &nbsp; Apps are available
                    for network and molecular profiling analyses, new layouts, additional file format support,
                    scripting, and connection with databases. &nbsp; They may be developed by anyone using the
                    Cytoscape open API based on <a href="http://www.java.com/en/" target="_blank">Java&#8482;</a>
                    technology and App community development is encouraged. Most of the Apps are
                    freely available from <a href="http://apps.cytoscape.org/">Cytoscape App Store</a>.
                </p>
            </div>
        </div>
    </div>
</div>

<div class="belt black image-wide">
    <div class="container">
        <img src="images/irefindex.png" title="Visualization of human interactome from iRefindex data set."
             style="width: 100%"/>
    </div>
</div>

<div class="narrow light-gray">
    <div class="container">

        <div class="row centering">
            <h1>Cytoscape Family</h1>
        </div>
    </div>
</div>

<div class="narrow light-gray">
    <div class="container">
        <div class="row">
            <div class="col-xs-12 col-md-5">
                <h1>Cytoscape 3</h1>

                <p class="lead">
                    <i>Cytoscape 3</i> is the mainstream version of Cytoscape with modular architecture.
                    It is designed for long-term maintainability and it replaced 2.x series.
                    New features, including new user interfaces, advanced visualization functions, headless
                    (command-line)
                    distribution, and multiple rendering engine support, will be released
                    for this version. As of November 2014, Cytoscape community is still in the process of transition
                    from 2.x to 3.x. Many Apps are now available for Cytoscape 3, and third party App developers
                    are working hard to port 2.x Apps to 3.x series.
                </p>
            </div>
            <div class="col-xs-12 col-md-7">
                <img src="images/screenshots/31_original.png" width="100%">
            </div>
        </div>
    </div>
</div>

<div class="narrow light-gray" id="js-section">
    <div class="container">
        <div class="row">
            <div class="col-xs-12">
                <h3>Cytoscape.js</h3>

                <p class="lead">
                    <a href="http://cytoscape.github.com/cytoscape.js/index.html">Cytoscape.js</a> is a successor of
                    <a href="http://cytoscapeweb.cytoscape.org">Cytoscape Web</a>. Cytoscape.js is a <strong>JavaScript
                    library</strong> for network visualization and analysis, and is <strong>NOT a complete web
                    application</strong>. Cytoscape 3 and cytoscape.js share design-level concepts, such as Visual
                    Styles,
                    but their code bases are completely independent to each other. It is designed to be a building block
                    for
                    complex
                    data visualization web applications for the HTML5 era.
                    The Cytoscape core developer team is trying to improve interoperability of Cytoscape 3 and
                    Cytoscape.js. 
                    From
                    Cytoscape 3.2, users can export network visualizations as interactive web application using
                    Cytoscape.js.
                </p>
            </div>
        </div>
        <div class="row">
            <div class="col-xs-12">
                <div class="cyjs"></div>
                <p class="lead">Interactive visualization with Cytoscape.js (touch me!)</p>
            </div>
        </div>

        <div class="row">
            <a name="which_cytoscape"></a>

            <div class="col-xs-12 col-md-5">
                <h3>Cytoscape 2</h3>

                <p>
                    <strong>Cytoscape 2.x</strong> is a legacy version of Cytoscape. This version is in
                    maintenance mode and new features will not be added by the core developers. If you need to
                    use a specific plugin in your workflow, you may need to use this version.
                </p>
            </div>
            <div class="col-xs-12 col-md-7">

                <h3>Which one is the right one for me?</h3>

                <ul class="list-unstyled">
                    <li>
                        <strong>Users, Casual Users, and Power Users:</strong> Use Cytoscape v3.x -- v2.x has been
                        in the field for several years and will be supported into the future, but over
                        time, v3.x and successors will be better supported. Use v2.x if you have used
                        v2.x in the past and rely on a plugin not supported as a v3.x app in the App
                        Store â€“ keep checking the App Store for new arrivals.
                    </li>
                    <li>
                        <strong>App Developers:</strong> Please port your 2.x Apps to 3.x series. We recommend to use
                        3.x for new plugins development. <a href="http://chianti.ucsd.edu/cytoscape-3.1.0/API/">3.x
                        API</a>
                        is stable enough to develop new Apps,
                        but if you notice any design flaws or bugs,
                        <a href="https://groups.google.com/forum/?fromgroups#!forum/cytoscape-discuss">
                            please let us know</a>.
                    </li>
                    <li>
                        <strong>Web Application Developers:</strong> You can use cytoscape.js as network data
                        visualization
                        engine for your web application. It is a pure JavaScript library and no plugin
                        is required for the web browsers. In the long term, this library and other
                        server-side applications replaces Java Web Start version of Cytoscape.
                    </li>
                </ul>
            </div>
        </div>
    </div>
</div>


<div class="narrow white">
    <div class="container centering">
        <h1>Cytoscape Core Features</h1>
    </div>
</div>

<div class="belt white">
    <div class="container">
        <div class="row">
            <div class="col-md-4 panel">
                <h4>Supports Many Standards</h4>
                <img src="images/intro_3.png"/>

                <p>
                    Cytoscape supports a lot of standard network and annotation file formats including: SIF
                    (Simple Interaction Format), <a href="http://www.infosun.fim.uni-passau.de/Graphlet/GML/"> GML</a>,
                    <a
                            href="http://cgi5.cs.rpi.edu/research/groups/pb/punin/public_html/XGMML/">
                        XGMML</a>, <a href="http://www.biopax.org/">BioPAX</a>,
                    <a href="http://www.psidev.info/index.php?q=node/60">PSI-MI</a>,
                    <a href="http://graphml.graphdrawing.org/">GraphML</a>,
                    <a href="http://www.kegg.jp/kegg/xml/">KGML (KEGG XML)</a>,
                    <a href="http://sbml.org/Main_Page">SBML</a>, <a href="http://www.obofoundry.org/">OBO</a>,
                    and <a href="http://www.geneontology.org/GO.format.annotation.shtml">Gene Association</a>.
                    Delimited text files and MS Excel&#8482; Workbook are also supported and you can import
                    data files, such as expression profiles or <a href="http://www.geneontology.org">GO</a> annotations,
                    generated by other applications
                    or spreadsheet programs. Using this feature, you can load and save arbitrary attributes
                    on nodes, edges, and networks. For example, input a set of custom annotation terms for
                    your proteins, create a set of confidence values for your protein-protein interactions.
                </p>
            </div>
            <div class="col-md-4 panel">
                <h4>Web Service Clients</h4>

                <p>
                    <img src="images/intro_4.png"/>
                </p>

                <p>
                    Cytoscape works as a web service client. This means Cytoscape can directly connect to
                    the external public databases and imports network and annotation data. Currently, <a
                        href="http://www.pathwaycommons.org/pc/">Pathway Commons</a>, <a
                        href="http://www.ebi.ac.uk/intact/site/index.jsf">IntAct</a>, <a
                        href="http://www.biomart.org/">BioMart</a>,
                    and <a href="http://www.ncbi.nlm.nih.gov/sites/entrez?db=gene">NCBI Entrez Gene</a>
                    are supported. And we continue to
                    develop new service clients for popular databases.
                </p>
            </div>
            <div class="col-md-4 panel">
                <h4>Interoperability</h4>


                <img src="images/logo/cy3logoOrange.svg" width="250px"/>
                <img src="https://www.python.org/static/community_logos/python-logo-generic.svg" width="250px"/>


                <p>
                    Since Cytoscape supports import/export standard file formats, you can easily put Cytoscape
                    into your workflow. For example, if you have a network data generated by
                    <a href="http://cneurocvs.rmki.kfki.hu/igraph/">igraph</a> or
                    <a href="http://www.bioconductor.org/packages/release/GraphsAndNetworks.html">Bioconductor</a>,
                    Cytoscape can load the file as a text table and you can export it in PSI-MI format for
                    other bioinformatics tools or your own applications/scripts.
                </p>
            </div>
        </div>
        <div class="row">
            <div class="col-md-4 panel">
                <h4>Session File</h4>

                <p>
                    <img src="images/cy3session.png"/>
                </p>

                <p>
                    You can save your work by a single click. All of settings, data files, and visualizations
                    are packed in a session file. It is called <i>Cytoscape Session (.cys) file</i>.
                    Cytoscape Session file includes networks, attributes (for node/edge/network), Desktop
                    states (selected/hidden nodes and edges, window sizes), Properties, some plugin states,
                    and Visual Styles.
                </p>
            </div>
            <div class="col-md-4 panel">
                <h4>Layout</h4>

                <p>
                    <img src="images/intro_9.png"/>
                </p>

                <p>
                    Layout networks in two dimensions. &nbsp;A variety of layout algorithms are available,
                    including cyclic, tree, force-directed, edge-weight, and yFiles Organic layouts. You can
                    also use Manual Layout tools similar to other graphics application user interface.
                </p>
            </div>
            <div class="col-md-4 panel">
                <h4>Image Export</h4>

                <img src="images/logo/adobe21.svg" width="200"/>

                <p>
                    You can export networks as publishiable-quality images. Supports PDF, PS, SVG, PNG,
                    JPEG, and BMP files. Vector images (PDF and PS) can be modified by other application
                    such as Adobe Illustrator for further enhancements.
                </p>
            </div>
        </div>
        <div class="row">
            <div class="col-md-6 panel">
                <h4>VizMapper</h4>

                <p>
                    <img src="images/screenshots/31_vizmap.png"/>
                </p>

                <p>
                    Customize network data display using powerful
                    <a href="http://wiki.cytoscape.org/Cytoscape_User_Manual/Visual_Styles">
                        <strong><i>VisualStyles</i></strong></a>.
                    View a superposition of gene expression ratios and p-values on the network. &nbsp;
                    Expression data can be mapped to node color, label, border thickness, or border color,
                    etc. according to user-configurable colors and visualization schemes.
                </p>
            </div>
            <div class="col-md-6 panel">
                <h4>Filter</h4>

                <p>
                    <img src="images/screenshots/31_filter.png" width="100%"/>
                </p>

                <p>
                    <strong>Filter the network</strong> to select subsets of nodes and/or interactions based
                    on the current data. &nbsp;For instance, users may select nodes involved in a threshold
                    number of interactions, nodes that share a particular GO annotation, or nodes whose gene
                    expression levels change significantly in one or more conditions according to p-values
                    loaded with the gene expression data. You can create new networks from the filtering result. </p>
            </div>
        </div>
        <div class="row">
            <div class="col-md-6 panel">
                <h4>Search</h4>

                <p>
                    <img src="images/intro_17.png"/>
                </p>

                <p>
                    Search target nodes and edges by Quick Find or
                    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2427162/">Enhanced Search</a>.
                    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2427162/table/T1/">
                        Lucene Syntax</a> is
                    supported by <i>Enhanced Search Plugin (ESP)</i> for arbitrary complex queries.
                </p>
            </div>
            <div class="col-md-6 panel">
                <h4>Browsing</h4>

                <p>
                    <img src="images/intro_11.png"/>
                </p>

                <p>
                    Zoom in/out and pan for browsing the network. Use the network manager to easily organize
                    multiple networks. And this structure can be saved in a session file. Use the <i>bird's
                    eye view</i> to easily navigate large networks. Easily navigate large networks (100,000+ nodes and
                    edges) by efficient rendering engine.
                </p>
            </div>
        </div>
        <div class="row">
            <div class="col-md-4 panel">
                <h4>Find Modules/ Clusters</h4>

                <p>
                    <strong>Find active subnetworks/pathway modules</strong>. The network is screened against gene
                    expression
                    data to identify connected sets of interactions, i.e. <i>interaction subnetworks</i>, whose genes
                    show
                    particularly high levels of differential expression. &nbsp;The interactions contained in each
                    subnetwork
                    provide hypotheses for the regulatory and signaling interactions in control of the observed
                    expression
                    changes. <strong>Find clusters (highly interconnected regions)</strong> in any network loaded into
                    Cytoscape. Depending on the type of network, clusters may mean different things. For instance,
                    clusters in a
                    protein-protein interaction network have been shown to be protein complexes and parts of pathways.
                    Clusters
                    in a protein similarity network represent protein families.
                </p>

                <p>
                    <img src="images/intro_14.png" width="100%"/>
                </p>

                <p>
                    <img src="images/intro_15.png" width="100%"/>
                </p>
            </div>
            <div class="col-md-4 panel">
                <h4>App Manager and <i>App Store</i></h4>
                <img src="images/appstore.png">

                <p>
                    <strong><a href="http://apps.cytoscape.org">Apps</a></strong> are available for network and
                    molecular
                    profile analysis. Cytoscape is a software written in Java and you can write your own
                    App for data analysis, import, and visualization by writing Java code. More Apps are
                    available at the <a href="http://apps.cytoscape.org">Cytoscape App Store</a>.
                    You can install most of the Apps by just one click from <i>App Manager</i>, or directly install
                    from the App Store (this is a Cytoscape 3 feature).
                </p>
            </div>
            <div class="col-md-4 panel">
                <h4>Multilingual</h4>
                <img src="images/intro_13.png" width="100%"/>

                <p>
                    You can use your language for the data files. Most of the features in Cytoscape supports
                    languages other than English, including Eastern Asian languages using two-byte characters.
                </p>
            </div>
        </div>
    </div>
</div>


<div class="narrow light-gray">
    <div class="container">

        <div class="row centering">
            <h1>History and Future</h1>
        </div>
    </div>
</div>
<div class="narrow light-gray">
    <div class="container">
        <div class="row">
            <div class="col-xs-12">
                <p class="lead">
                    Cytoscape was initially made public in July, 2002 (v0.8); the second release (v0.9) was in
                    November, 2002 and v1.0 was released in March 2003. Version 1.1.1 is the last stable
                    release for the 1.0 series. Latest version of Cytoscape 2.x series is 2.8.3.
                </p>

                <p class="lead">
                    Cytoscape 3.0, the first release of more modularized, scalable version
                    of Cytoscape, is now available for advanced users and App developers. For more information, please
                    visit <a href="cy3.html">this page</a>.
                </p>
            </div>
        </div>
    </div>
</div>

<footer></footer>

<script src="js/load_cyjs.js"></script>
<script src="js/setup_page.js"></script>

</body>
</html>
